pkg/controller/windowsmachineconfig/nodeconfig/network.go

```diff
@@ -0,0 +1,134 @@
+package nodeconfig
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/clusternetwork"
+	"github.com/pkg/errors"
+)
+
+// cniConf contains the structure of the CNI template
+type cniConf struct {
+	CNIVersion   string       `json:"cniVersion"`
+	Name         string       `json:"name"`
+	Type         string       `json:"type"`
+	Capabilities capabilities `json:"capabilities"`
+	IPAM         ipam         `json:"ipam"`
+	Policies     policies     `json:"policies"`
+}
+
+// nested structs required for creating cniConf struct
+type capabilities struct {
+	DNS bool `json:"dns"`
+}
+
+type ipam struct {
+	Type   string `json:"type"`
+	Subnet string `json:"subnet"`
+}
+
+type policies []struct {
+	Name  string `json:"name"`
+	Value value  `json:"value"`
+}
+
+type value struct {
+	Type              string   `json:"Type"`
+	ExceptionList     []string `json:"ExceptionList,omitempty"`
+	DestinationPrefix string   `json:"DestinationPrefix,omitempty"`
+	NeedEncap         bool     `json:"NeedEncap"`
+}
+
+// network struct contains the node network information
+type network struct {
+	// hostSubnet holds the node host subnet value
+	hostSubnet string
+}
+
+// newNetwork returns a pointer to the network struct
+func newNetwork() *network {
+	return &network{}
+}
+
+// setHostSubnet sets the value for hostSubnet field in the network struct
+func (nw *network) setHostSubnet(hostSubnet string) error {
+	if hostSubnet == "" || clusternetwork.ValidateCIDR(hostSubnet) != nil {
+		return errors.Errorf("error receiving valid value for node hostSubnet")
+	}
+	nw.hostSubnet = hostSubnet
+	return nil
+}
+
+// cleanupTempConfig cleans up the temporary CNI directory and config file created
+func (nw *network) cleanupTempConfig(configFile string) error {
```

Comment:
This should be a generic clean up function. It should not take `configFile` as input but `configFile` can become the property of the struct. With the way it stands, you can pass any file into this function and it will be deleted.

Reply:
I am okay leaving it as it is as I realize this is local and not an exported function.